lib/cosmos/dart/processes/dart_worker.rb

```diff
@@ -0,0 +1,228 @@
+              begin
+                switch_and_get_system_config(system_config_name)
+              rescue
+                Cosmos::Logger.error("Could not load system_config: #{system_config_name}")
+                ple.decom_state = PacketLogEntry::NO_CONFIG
+                ple.save
+                Cosmos::Logger.error("PLE:#{ple.id}:#{ple.decom_state_string}")
+                next
+              end
+
+              # Read the actual packet
+              packet = read_packet_from_ple(ple)
+              unless packet
+                ple.decom_state = PacketLogEntry::NO_PACKET
+                ple.save
+                Cosmos::Logger.error("PLE:#{ple.id}:#{ple.decom_state_string}")
+                next
+              end
+
+              packet_id = ple.packet_id
+
+              # Have the packet and the system configuration - Need to lookup the data on where to decom to
+              packet_config = PacketConfig.where("packet_id = ? and name = ?", packet_id, packet.config_name).first
+              begin
+                unless packet_config
+                  Cosmos::Logger.info("Creating PacketConfig: #{packet.config_name}...")
+                  packet_config = PacketConfig.create(:packet_id => packet_id, :name => packet.config_name, :first_system_config_id => system_config.id)
+                  setup_packet_config(packet, packet_id, packet_config)
+                  Cosmos::Logger.info("Successfully Created PacketConfig: #{packet.config_name}")
+                end
+              rescue => err
+                Cosmos::Logger.error(err.formatted)
+                # Another thread probably already created it - Try to get it one more time
+                packet_config = PacketConfig.where("packet_id = ? and name = ?", packet_id, packet.config_name).first
+              end
+
+              unless packet_config
+                ple.decom_state = PacketLogEntry::NO_PACKET_CONFIG
+                ple.save
+                Cosmos::Logger.error("PLE:#{ple.id}:#{ple.decom_state_string}")
+                next
+              end
+
+              first = true
+              while !packet_config.ready
+                if first
+                  ready_wait_start = Time.now
+                  Cosmos::Logger.info("Waiting for PacketConfig:#{packet_config.id}...")
+                  first = false
+                end
+                sleep(1)
+                packet_config.reload
+
+                if (Time.now - ready_wait_start) > 60
+                  Cosmos::Logger.fatal("Timeout waiting for ready on PacketConfig:#{packet_config.id}")
+                  exit(0)
+                end
+              end
+
+              # Ready to decom!
+              ple.decom_state = PacketLogEntry::IN_PROGRESS
+              ple.save
+              decom_packet(ple, packet, packet_id, packet_config.id)
+              ple.decom_state = PacketLogEntry::COMPLETE
+              ple.save
+
+              if ple.decom_state == PacketLogEntry::COMPLETE
```

Comment:
This check seems unneccessary. We just set decom_state to COMPLETE so how could it be anything else?

Reply:
Agreed